File: snippets/fsharp/System/UInt32/MaxValue/MaxValue1.fs
open System 

// <Snippet1>
let longValue = Int64.MaxValue / 2L

if longValue <= int64 UInt32.MaxValue && longValue >= int64 UInt32.MinValue then
    let integerValue = uint longValue 
    printfn $"Converted long integer value to {integerValue:n0}." 
else
    let rangeLimit, relationship = 
        if longValue > int64 UInt32.MaxValue then
            UInt32.MaxValue, "greater" 
        else
            UInt32.MinValue, "less" 

    printfn $"Conversion failure: {longValue:n0} is {relationship} than {rangeLimit:n0}"
// </Snippet1>

File: snippets/csharp/System/UInt32/MaxValue/MaxValue1.cs
using System;

public class ULongRangeExample
{
    public static void Main()
    {
        // <Snippet1>
        long longValue = long.MaxValue / 2;
        uint integerValue;

        if (longValue <= uint.MaxValue &&
            longValue >= uint.MinValue)
        {
            integerValue = (uint)longValue;
            Console.WriteLine($"Converted long integer value to {integerValue:n0}.");
        }
        else
        {
            uint rangeLimit;
            string relationship;

            if (longValue > uint.MaxValue)
            {
                rangeLimit = uint.MaxValue;
                relationship = "greater";
            }
            else
            {
                rangeLimit = uint.MinValue;
                relationship = "less";
            }

            Console.WriteLine($"Conversion failure: {longValue:n0} is {relationship} than {rangeLimit:n0}");
        }
        // </Snippet1>
    }
}
